docs/dev/Lua API.rst

```diff
@@ -3928,6 +3928,211 @@ Examples:
+
+* ``dateTime:dayOfMonthWithSuffix()``
+
+  Returns *day of month, as a string, including its ordinal suffix*.
+
+* ``dateTime:month()``
+
+  Returns *current month, as a number, starting from one*.
+
+* ``dateTime:setMonth(month)``
+
+  Sets the current month starting from one.
+  Returns *DateTime*.
+
+* ``dateTime:nameOfMonth()``
+
+  Returns *name of the current month*.
+
+* ``dateTime:dayOfYearTick()``
+
+  Returns *days from ticks since beginning of current year starting from zero*.
+
+* ``dateTime:dayOfMonthTick()``
+
+  Returns *days from ticks since the beginning of current month starting from zero*.
+
+* ``dateTime:dayTick()``
+
+  Returns *ticks since beginning of the current day starting from zero*.
+
+* ``dateTime:setDayTick(ticks)``
+
+  Sets the current day's tick count.
+
+* ``dateTime:monthOfYearTick()``
+
+  Returns *months from ticks since the beginning of the current year starting from zero*.
+
+* ``dateTime:monthTick()``
+
+  Returns *ticks since the beginning of a month*.
+
+* ``dateTime:daysToTicks(days)``
+
+  Returns *days converted to ticks*.
+
+* ``dateTime:monthsToTicks(months)``
+
+  Returns *months converted to ticks*.
+
+* ``dateTime:addTicks(ticks)``
+
+  Adds *ticks*, normalizing if needed.
+  Returns *DateTime*.
+
+* ``dateTime:addDays(days)``
+
+  Adds *days*, normalizing if needed.
+  Returns *DateTime*.
+
+* ``dateTime:addMonths(months)``
+
+  Adds *months*, normalizing if needed.
+  Returns *DateTime*.
+
+* ``dateTime:addYears(years)``
+
+  Adds *years* to dateTime. Returns *DateTime*.
+
+* ``dateTime:time()``
+
+  Returns three integers: *hours (24 hour time)*, *minutes*, *seconds* of the current day.
+
+* ``dateTime:setTime(hours, minutes, seconds)``
+
+  Sets *hours*, *minutes*, *seconds* of the current day.
+  returns *DateTime*.
+
+* ``dateTime:toDuration()``
+
+  Returns a *Duration object*.
+
+* ``dateTime:normalize()``
+
+  Ensure's year_tick is within bounds of a year, otherwise
+  adjusts year and year_tick accordingly. Note: should only
+  call this function when directly modifying year_tick.
+
+* ``dateTime:setTickRate(ticks_per_day)``
+
+  Sets the tick rate from the provided ticks_per_day. Returns *DateTime*.
+  If ticks_per_day is equal to the value of df.game_mode.DWARF (currently 0)
+  or df.game_mode.ADVENTURE (currently 1) then the tick rate will be set to the
+  respective rate. *Defaults to fortress mode*.
+
+* ``DateTime.now([game_mode])``
+
+  Returns *DateTime* initialized from the current game time.
+  If game_mode is not given. It detects the current game mode.
+  When in *adventure mode*, DateTime is initialized by adventure mode time.
+  Otherwise *defaults to fortress mode* time. This function sets tick rate by game mode.
+  If you need a custom tick rate, call dateTime:setTickRate(custom_ticks_per_day).
+
+Duration Class
+==============
+
+Inherits DateTime's constructor and methods. Used for measuring the difference
+between DateTime and Duration objects.
```

Comment:
A lot of the DateTime instance functions don't really make sense for a Duration class. Would this be better the other way? Duration as the superclass and DateTime as the subclass?

Reply:
The other way works. Some calendrical functions will need hoisting to Duration. And the addDays(), addMonths(), etc ones would probably be useful for Duration as well. In fact, it's not uncommon from implementations that I've looked at, to expose duration-style functions in their date/time objects.

It's either that, or DwarfCalendar can be the super class for common calendrical logic. I don't know if the calendar is actually moddable.  

Additionally, Duration wouldn't need to know about the calendar at all if it didn't include magnitudes of months and years. Just some thoughts.

Thanks for looking at this.